src/notifications/ldn.ts

```diff
@@ -160,32 +159,30 @@ export async function unstable_sendNotificationToInbox(
 /**
  * Discovers the inbox of the provided target resource, and then sends the provided notification to that inbox.
  * Fails if no inbox is discovered.
- * @param notification The content of thoe notification
- * @param receiver The target resource (e.g. a WebID)
+ *
+ * @param notification The content of the notification.
+ * @param receiver The URL of the target resource.
  * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
- * @returns A Promise resolving to a [[LitDataset]] containing the stored data linked to the new notification Resource, or rejecting if saving it failed.
+ * @returns A Promise resolving to a [[Thing]] containing the stored data linked to the new notification Resource, or rejecting if saving it failed.
  */
 export async function unstable_sendNotification(
-  notification: LitDataset & Partial<{ notification: UrlString | LocalNode }>,
-  receiver: Url | UrlString,
+  notification: Thing,
+  receiver: Url | UrlString | ThingPersisted,
   options: Partial<
     typeof internal_defaultFetchOptions
   > = internal_defaultFetchOptions
 ) {
-  let notificationToSend = notification;
-  const inbox = await unstable_fetchInbox(receiver, options);
+  // TODO: Change `fetchInbox` to take a `ThingPersisted` and just read the inbox from it directly:
+  const receiverIri = internal_toIriString(toNode(receiver));
+  const inbox = await unstable_fetchInbox(receiverIri, options);
   if (inbox === null) {
-    throw new Error(
-      `No inbox discovered for resource [${internal_toString(receiver)}]`
-    );
+    throw new Error(`No inbox discovered for Resource [${receiverIri}]`);
   }
-  if (notification.notification !== undefined) {
-    let notificationThing = getThingOne(
-      notification,
-      notification.notification
-    );
-    notificationThing = setIri(notificationThing, as.target, receiver);
-    notificationToSend = setThing(notificationToSend, notificationThing);
-  }
-  return unstable_sendNotificationToInbox(notificationToSend, inbox, options);
+
+  notification = setIri(notification, as.target, receiver);
```

Comment:
Yeah it's an in-between - we don't mutate the actual argument, but we do re-assign the variable. Given that it stood out to you, I'll change it to explicitly initialise a new variable.

Reply:
To be fair, it only stood out because I literally changed that piece of code yesterday and created the `notificationToSend` variable exactly for this purpose, so I really don't mind this staying this way if it's not against guidelines/practices.

Reply:
12985fac62f8fccc59179f98a19eb0e4d49dd1a2

Reply:
It depends on the guidelines, so better safe than sorry :) (Also I've already done it :P )